clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1948,6 +1971,7 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
       if (LC.capturesThis() && LC.isImplicit())
         Diag(LC.getLocation(), diag::err_implicit_this_capture);
   }
+  diagKernelCallerSpecConformance(*this, KernelCallerFunc);
```

Comment:
The comment I had made in the past was to move the checks from 1969 into their own function as soon as we had a 2nd thing to check.  So please make 1 function that checks both:

checkKernel<somethingorother>?  Anyone wanna bikeshed for me? @elizabethandrews  @premanandrao ?

Reply:
checkKernelParameters?

Reply:
checkKernelAndCaller ?

Reply:
checkKernelCompat ?

Reply:
> checkKernelParameters?

Its not really the Parameters, to the kernel, its the kernel object itself (plus the caller).  


> checkKernelAndCaller ?
Probably good enough for now.  Presumably someone will come up with a better idea in the future and we can rename it.

Reply:
done.